kotest/src/test/kotlin/com/baeldung/kotest/beforetest/BeforeTestSamples.kt

```diff
@@ -0,0 +1,48 @@
+package com.baeldung.kotest.beforetest
+
+import io.kotest.core.listeners.BeforeInvocationListener
+import io.kotest.core.spec.IsolationMode
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.TestCase
+import io.kotest.matchers.shouldBe
+
+class BeforeTestSamples : FunSpec({
+
+    val userRepository = UserRepository()
+
+    beforeEach {
+        userRepository.addUser(User(1, "Admin"))
+    }
+
+    isolationMode = IsolationMode.InstancePerTest
```

Comment:
Let's flip the order of these.

```suggestion
    isolationMode = IsolationMode.InstancePerTest

    beforeEach {
        userRepository.addUser(User(1, "Admin"))
    }
```